src/master-playlist-controller.js

```diff
@@ -361,6 +370,32 @@ export class MasterPlaylistController extends videojs.EventTarget {
         } else {
           addSeekableRange();
         }
+
+        buffered = this.tech_.buffered();
+        currentTime = this.tech_.currentTime();
+
+        if (buffered.length) {
+          lastBufferedEnd = buffered.end(buffered.length - 1);
+        }
+        bufferedTime = lastBufferedEnd - currentTime;
+
+        // the end of the current playlist
+        playEnd = this.playlistEnd(updatedPlaylist);
+
+        // the time between end of the playlist and the end of the buffered
+        endTime = playEnd - lastBufferedEnd;
+
+        if (this.seekable().length) {
+          seekableEnd = this.seekable().end(0);
+        }
+
+        // blacklist this playlist if the playhead is at the end of the buffer
+        // and the buffered data ends at the end of the last segment in the playlist
+        if (bufferedTime <= Ranges.TIME_FUDGE_FACTOR &&
+            endTime <= Ranges.TIME_FUDGE_FACTOR &&
+            currentTime > seekableEnd) {
```

Comment:
With `lastBufferedEnd` for example, you declare `lastBufferedEnd` at the start of this function, so it is `undefined`. The only location `lastBufferedEnd` is assigned a value is on line 378, which is inside the `if (buffered.length)` check. If `buffered.length === 0` then `lastBufferedEnd` will still be `undefined` and any calculation involving `lastBufferedEnd` e.g. `endTime = playEnd - lastBufferedEnd` will result in `NaN`.

With that said, looking at this again, it may be ok for this to happen. Since any comparison with `NaN` will return false, `seekable.length === 0` or `buffered.length === 0` probably means we don't have enough information to know if the playlist stopped updating and probably shouldn't blacklist it. This will require testing to see if we can create a scenario in which those values result in `NaN` but we still want to blacklist